hummingbot/connector/exchange/btc_markets/btc_markets_exchange.py

```diff
@@ -0,0 +1,966 @@
+        # take `in_flight_orders` snapshot during `_update_balances` (see #2224)
+        self._in_flight_orders_snapshot = {k: copy.copy(v) for k, v in self._in_flight_orders.items()}
+        self._in_flight_orders_snapshot_timestamp = btc_markets_utils.get_ms_timestamp()
+
+    async def _update_order_status(self):
+        """
+        Calls REST API to get status update for each in-flight order.
+        """
+
+        last_tick = int(self._last_poll_timestamp / self.UPDATE_ORDER_STATUS_MIN_INTERVAL)
+        current_tick = int(self.current_timestamp / self.UPDATE_ORDER_STATUS_MIN_INTERVAL)
+
+        if current_tick > last_tick and len(self._in_flight_orders) > 0:
+            tracked_orders = list(self._in_flight_orders.values())
+            tasks = []
+            for tracked_order in tracked_orders:
+                order_id = await tracked_order.get_exchange_order_id()
+                tasks.append(self._api_request("GET",
+                                               f"{Constants.ORDERS_URL}/{order_id}",
+                                               {},
+                                               # Constants.ORDERS_URL,
+                                               # {"id": order_id},
+                                               True))
+            self.logger().debug(f"Polling for order status updates of {len(tasks)} orders.")
+            responses = await safe_gather(*tasks, return_exceptions=True)
+            for response in responses:
+                if isinstance(response, Exception):
+                    raise response
+                if "status" not in response:
+                    self.logger().info(f"_update_order_status result not in resp: {response}")
+                    continue
+
+                if "trades" in response:
+                    await self._process_trade_message(response)
+
+                await self._process_order_message(response)
+
+    async def _process_order_message(self, order_msg: Dict[str, Any]):
+        """
+        Updates in-flight order and triggers cancellation or failure event if needed.
+        :param order_msg: The order response from either REST or WebSocket but WS doesn't return clientOrderId on
+        orderChange events. https://api.btcmarkets.net/doc/v3#section/Order-life-cycle-events
+        """
+
+        if "clientOrderId" in order_msg:
+            client_order_id = order_msg["clientOrderId"]
+            if client_order_id not in self._in_flight_orders.get(client_order_id):
+                self.logger().debug(f"Order not currently tracked: {order_msg['clientOrderId']}")
+                return
+            else:
+                tracked_order = self._in_flight_orders[client_order_id]
+                tracked_order.last_state = order_msg["status"]
+
+        if "orderId" in order_msg:
+            orderId = order_msg["orderId"]
+            for key, value in self._in_flight_orders.items():
+                inflightorderId = value.exchange_order_id
```

Comment:
The exchange order id should always be accessed using the function `get_exchange_order_id()`, handling the possible TimeoutError